src/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -3078,6 +3078,16 @@ public boolean isAutoSnapshotEnabled()
         return metadata().params.allowAutoSnapshot && DatabaseDescriptor.isAutoSnapshot();
     }
 
+    public DurationSpec.IntSecondsBound getAutoSnapshotTtl()
+    {
+        // If it is not set as a table parameter (which is by default 0s), then take the default value from descriptor.
+        // It is possible that it is unset in descriptor, which again defaults to 0s.
+        // When SnapshotManifest receives ttl with null or 0s, it will not write expiration time to manifest file
+        // Hence, table parameter has precedence over descriptor value.
+        DurationSpec.IntSecondsBound ttlFromTable = metadata().params.autoSnapshotTtl;
```

Comment:
`autoSnapshotTtl` is not null because we need to have non-null value when e.g. describing a table. It is way easier to work with non-null values. If `DatabaseDescriptor.getAutoSnapshotTtl` was null, we would have non-null ttl from params which might be of value 0s or we would possibly return null from descriptor.

I think that this method should return non-null in every case and making special logic about nullity is cumbersome. Hence I prefer to have it non-null every time.